lib/coinbase/address.rb

```diff
@@ -1,271 +1,31 @@
-    # @param asset_id [Symbol] The Asset to retrieve the balance for
-    # @return [BigDecimal] The balance of the Asset
-    def balance(asset_id)
-      response = Coinbase.call_api do
-        addresses_api.get_address_balance(wallet_id, id, Coinbase::Asset.primary_denomination(asset_id).to_s)
-      end
-
-      return BigDecimal('0') if response.nil?
-
-      Coinbase::Balance.from_model_and_asset_id(response, asset_id).amount
-    end
-
-    # Transfers the given amount of the given Asset to the specified address or wallet.
-    # Only same-network Transfers are supported.
-    # @param amount [Integer, Float, BigDecimal] The amount of the Asset to send.
-    # @param asset_id [Symbol] The ID of the Asset to send. For Ether, :eth, :gwei, and :wei are supported.
-    # @param destination [Wallet | Address | String] The destination of the transfer. If a Wallet, sends to the Wallet's
-    #  default address. If a String, interprets it as the address ID.
-    # @return [Coinbase::Transfer] The Transfer object.
-    def transfer(amount, asset_id, destination)
-      destination_address, destination_network = destination_address_and_network(destination)
-
-      validate_can_transfer!(amount, asset_id, destination_network)
-
-      transfer = create_transfer(amount, asset_id, destination_address)
+    attr_reader :network_id, :id
 
-      # If a server signer is managing keys, it will sign and broadcast the underlying transfer transaction out of band.
-      return transfer if Coinbase.use_server_signer?
-
-      broadcast_transfer(transfer, transfer.transaction.sign(@key))
-    end
-
-    # Trades the given amount of the given Asset for another Asset.
-    # Only same-network Trades are supported.
-    # @param amount [Integer, Float, BigDecimal] The amount of the Asset to send.
-    # @param from_asset_id [Symbol] The ID of the Asset to trade from. For Ether, :eth, :gwei, and :wei are supported.
-    # @param to_asset_id [Symbol] The ID of the Asset to trade to. For Ether, :eth, :gwei, and :wei are supported.
-    # @return [Coinbase::Trade] The Trade object.
-    def trade(amount, from_asset_id, to_asset_id)
-      validate_can_trade!(amount, from_asset_id, to_asset_id)
-
-      trade = create_trade(amount, from_asset_id, to_asset_id)
-
-      # NOTE: Trading does not yet support server signers at this point.
-
-      payloads = { signed_payload: trade.transaction.sign(@key) }
-
-      payloads[:approve_tx_signed_payload] = trade.approve_transaction.sign(@key) unless trade.approve_transaction.nil?
-
-      broadcast_trade(trade, **payloads)
-    end
-
-    # Returns whether the Address has a private key backing it to sign transactions.
-    # @return [Boolean] Whether the Address has a private key backing it to sign transactions.
-    def can_sign?
-      !@key.nil?
+    # Returns a new Address object.
+    # @param network_id [Symbol] The Network ID
+    # @param id [String] The onchain Address ID
+    def initialize(network_id, id)
```

Comment:
Do we want user's to be able to initialize a `Coinbase::Address` itself in this world?

Reply:
Not really, Im thinking of updating the docs to say:

> This superclass should not be used explicitly, but instead the `DeveloperAddress` & `ExternalAddress` should be used